### PR TITLE
Refactor modules.config

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Applies ext.when at the subworkflow level instead of modules. 
- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test2samples,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -1127,34 +1118,35 @@ process {
...
         publishDir = [
             enabled: false
         ]
     }
 
     withName: '.*ANALYSE_MT:MERGE_ANNOTATE_MT:TABIX_TABIX_MT' {
-        ext.when = { params.mt_fasta_shift && params.mt_intervals && !(params.analysis_type == "wes") }
+        publishDir = [
+            enabled: false
+        ]
```
Comment (jemten) [inline conf/modules.config:1144]: is the publishDir directive necessary or is it just that you need something here in order for the switch to work?

**Review by jemten — COMMENTED** [review]
Nice work

File: conf/modules.config (hunk 1 above)
Comment (ramprasadn) [reply conf/modules.config:1144]: My idea is to have an entry for all the processes so this is sort of like explicitly stating the defaults. They don't have to be there

Comment (jemten) [reply conf/modules.config:1145]: I like it!

File: subworkflows/local/qc_bam.nf [hunk 2]
```diff
@@ -54,8 +54,8 @@ workflow QC_BAM {
         PICARD_COLLECTWGSMETRICS ( bam_bai, fasta, fai, intervals_wgs )
         PICARD_COLLECTWGSMETRICS_Y ( bam_bai, fasta, fai, intervals_y )
 
-        SENTIEON_WGSMETRICS ( bam_bai, fasta, fai)
-        SENTIEON_WGSMETRICS_Y ( bam_bai, fasta, fai)
+        SENTIEON_WGSMETRICS ( bam_bai, fasta, fai, intervals_wgs )
+        SENTIEON_WGSMETRICS_Y ( bam_bai, fasta, fai, intervals_y )
```
Comment (jemten) [inline subworkflows/local/qc_bam.nf:58]:
```suggestion
        SENTIEON_WGSMETRICS ( bam_bai, fasta, fai)
        SENTIEON_WGSMETRICS_Y ( bam_bai, fasta, fai)
```

**Review by jemten — APPROVED** [review]
Good stuff

**Pull request merged**